packages/@aws-cdk/region-info/README.md

```diff
@@ -0,0 +1,80 @@
+# AWS Region-Specific Information Directory
+## Usage
+Some information used in CDK Applications differs from one AWS region to
+another, such as service principals used in IAM policies, S3 static website
+endpoints, ...
+
+### Direct Access
+This library offers a primitive database of such information so that CDK
+constructs can easily access regional information. The `Facts` class provides a
+list of known fact names, which can then be used with the `RegionInfo` to
+retrieve a particular value:
+
+```ts
+import { Facts, RegionInfo } from '@aws-cdk/region-info';
+
+const codeDeployPrincipal = RegionInfo.find('us-east-1', Facts.servicePrincipal('codedeploy.amazonaws.com'));
+// => codedeploy.us-east-1.amazonaws.com
+
+const staticWebsite = RegionInfo.find('ap-northeast-1', Facts.s3StaticWebsiteEndpoint);
+// => s3-website-ap-northeast-1.amazonaws.com
+```
+
+### Token Access
+Often, CDK constructs are not able to access the name of the region for which a
+fact is requested (becase this information will not be available before the
+synthesis is underway). To make usage in such scenarios easier, the library
```

Comment:
What's the use case for tokens, not sure I fully understand?

Reply:
Often, I don't have the region available on the spot, and the resolution context of the token will allow me to get it without forcing each user to go dig for it.

Reply:
See my comment about moving tokens out of this layer